pkg/api/login.go

```diff
@@ -39,20 +39,45 @@ func LoginView(c *m.ReqContext) {
 		viewData.Settings["loginError"] = loginError
 	}
 
+	if tryOAuthAutoLogin(c) {
+		postLoginRedirect(c)
+		return
+	}
+
 	if !tryLoginUsingRememberCookie(c) {
 		c.HTML(200, ViewIndex, viewData)
+		postLoginRedirect(c)
```

Comment:
I don't think we should redirect in this case. This will cause double writes to the ResponseWriter as well

Reply:
Thanks for the hint. The problem should be fixed now.